src/main.cpp

```diff
@@ -2696,25 +2697,42 @@ static int64_t nTimeIndex = 0;
 static int64_t nTimeCallbacks = 0;
 static int64_t nTimeTotal = 0;
 
+/**
+ * Determine whether to do transaction checks when validating blocks.
+ * Returns `false` (allowing transaction checks to be skipped) only if all
+ * of the following are true:
+ *   - we're currently in initial block download
+ *   - the `-ibdskiptxverification` flag is set
+ *   - the block under inspection is an ancestor of the latest checkpoint.
+ */
+static bool IBDSkipTxVerification(const CChainParams& chainparams, const CBlockIndex* pindex) {
```

Comment:
So this is interesting - I originally had written this as you have suggested that I modify it here. However, I changed it because I wanted to make clear from the name that this was only relevant during initial block download, if at all. "ShouldCheck" seems a bit nebulous to me.

Reply:
To be the `IBD...` name could imply something that is only called during IBD, rather than something that can be called at any time but checks for IBD (and other things).